backend/api/app/application/services/user_service.py

```diff
@@ -76,3 +106,139 @@ def verifySSID(self, ssid: str, ip_address: str):
             return True
         else:
             return False
+        
+    def logout_user(self, ssid: str, ip_address: str):
+            try:
+                
+                ses_id = uuid.UUID(ssid)
+                session = SessionRepository.get_active_by_id(ses_id, ip_address)
+                
+                session.logged_out = True
+                SessionRepository.update(session)
+
+                return {"message": "User logged out successfully", "status": 200}
+            except Exception as ex:
+                return {"message": "An unexpected error occurred", "details": str(ex), "status": 500}
+            
+    def delete_user(self, user_id: uuid.UUID, password: str):
+        try:
+            user = User.query.get(user_id)
+            if not user:
+                return {"message": "User not found", "status": 404}
+
+            if not user.check_password(password):
+                return {"message": "Incorrect password", "status": 401}
+
+            user.is_deleted = True  
+            UserRepository.updateUser(user)
+
+            return {"message": "User account deleted successfully", "status": 200}
+
+        except Exception as ex:
+            return {"message": "An unexpected error occurred", "status": 500}
+        
+    def get_user_info(self, ssid: str, ip_address: str):
+        try:
+            ses_id = uuid.UUID(ssid)
+        except ValueError:
+            raise ValueError({"message": "Invalid session ID format.", "status": 400})
```

Comment:
We shouldn't expose data about our session, if session is not good that means user is not logged in and we should treat that as unauthenticated entrance.
Also, maybe there is no need for checking the format (besides if it's null or empty string) because we will check database for that session.